sdk/couchbase-core/src/mgmtx/user_json.rs

```diff
@@ -0,0 +1,173 @@
+use crate::mgmtx::error;
+use crate::mgmtx::user::{
+    Group, Origin, Role, RoleAndDescription, RoleAndOrigins, User, UserAndMetadata,
+};
+use chrono::DateTime;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialOrd, PartialEq)]
+pub struct RoleJson {
+    pub role: String,
+    #[serde(rename = "bucket_name", skip_serializing_if = "Option::is_none")]
+    pub bucket: Option<String>,
+    #[serde(rename = "scope_name", skip_serializing_if = "Option::is_none")]
+    pub scope: Option<String>,
+    #[serde(rename = "collection_name", skip_serializing_if = "Option::is_none")]
+    pub collection: Option<String>,
+}
+
+#[derive(Deserialize, Debug, Clone, PartialOrd, PartialEq)]
+pub struct RoleAndDescriptionJson {
+    #[serde(flatten)]
+    pub role_json: RoleJson,
+    #[serde(rename = "name")]
+    pub display_name: String,
+    #[serde(rename = "desc")]
+    pub description: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialOrd, PartialEq)]
+pub struct OriginJson {
+    #[serde(rename = "type")]
+    pub origin_type: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialOrd, PartialEq)]
+pub struct RoleAndOriginsJson {
+    #[serde(flatten)]
+    pub role_json: RoleJson,
+    pub origins: Vec<OriginJson>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialOrd, PartialEq)]
+pub struct GroupJson {
+    pub id: String,
+    pub description: Option<String>,
+    pub roles: Vec<RoleJson>,
+    #[serde(rename = "ldap_group_ref", skip_serializing_if = "Option::is_none")]
+    pub ldap_group_reference: Option<String>,
+}
+
+#[derive(Deserialize, Debug, Clone, PartialOrd, PartialEq)]
+pub struct UserJson {
```

Comment:
How does this handle new fields added by ns_server? Just wondering because there's at least three new fields which will be coming in 8.0